python/lancedb/table.py

```diff
@@ -490,6 +491,7 @@ def create_index(
             replace=replace,
         )
         self._reset_dataset()
+        EVENTS("create_index")
```

Comment:
Would it make more sense to have this as a decorator? Not a blocker for the first version though

Reply:
Yeah totally depends on out taste.